packages/datadog-plugin-cucumber/src/index.js

```diff
@@ -0,0 +1,95 @@
+const { relative } = require('path')
+
+const { SAMPLING_RULE_DECISION } = require('../../dd-trace/src/constants')
+
+const {
+  TEST_TYPE,
+  TEST_NAME,
+  TEST_SUITE,
+  TEST_STATUS,
+  getTestEnvironmentMetadata
+} = require('../../dd-trace/src/plugins/util/test')
+
+function setStatusFromResult (span, result) {
+  if (result.status === 1) {
+    span.setTag(TEST_STATUS, 'pass')
+  } else if (result.status === 2) {
+    span.setTag(TEST_STATUS, 'skip')
+    span.setTag('error.msg', 'skipped')
+  } else if (result.status === 4) {
+    span.setTag(TEST_STATUS, 'skip')
+    span.setTag('error.msg', 'not implemented')
+  } else {
+    span.setTag(TEST_STATUS, 'fail')
+    span.setTag('error.msg', result.message)
+  }
+}
+
+function createWrapRun (tracer, testEnvironmentMetadata, sourceRoot) {
+  return function wrapRun (run) {
+    return async function handleRun () {
+      const testName = this.pickle.name
+      const testSuite = relative(sourceRoot, this.pickle.uri)
+
+      const commonSpanTags = {
+        [TEST_TYPE]: 'test',
+        [TEST_NAME]: testName,
+        [TEST_SUITE]: testSuite,
+        [SAMPLING_RULE_DECISION]: 1,
+        ...testEnvironmentMetadata
+      }
+
+      const result = await tracer.trace(
+        'cucumber.test',
+        {
+          type: 'test',
+          resource: testName,
+          tags: commonSpanTags
+        },
+        async (span) => {
+          const runResult = await run.apply(this, arguments)
+          setStatusFromResult(span, this.getWorstStepResult())
+          return runResult
+        }
+      )
+      return result
+    }
+  }
+}
+
+function createWrapRunStep (tracer) {
+  return function wrapRunStep (runStep) {
+    return async function handleRunStep () {
+      const resource = arguments[0].isHook ? 'hook' : arguments[0].pickleStep.text
+      return tracer.trace(
+        'cucumber.step',
+        { type: 'test', resource: resource },
+        async (span) => {
+          const result = await runStep.apply(this, arguments)
```

Comment:
As explained above, this `await` can be replaced with a single `then` which will remove the need for all the other async/awaits in the function.

Reply:
I have removed the `await` 👍

Reply:
It looks like there are still several `async/await` in the code. The idea was to remove all of them to avoid creating tons of unnecessary promises. It's not as important for something like Cucumber compared to something running in production, but I think it's still worth it to keep this practice. This doesn't apply to our test code and async/await is fine there.